src/routes/v1/getTxInfo.ts

```diff
@@ -0,0 +1,111 @@
+import { asMaybe, asObject, asString } from 'cleaners'
+import Router from 'express-promise-router'
+
+import { reportsTransactions } from '../../indexApi'
+import { asDbTx, DbTx } from '../../types'
+import { EdgeTokenId } from '../../util/asEdgeTokenId'
+import { HttpError } from '../../util/httpErrors'
+import { trial } from '../../util/trail'
+
+const asGetTxInfoReq = asObject({
+  /**
+   * Prefix of the destination address.
+   * Minimum 3 character; Maximum 5 characters.
+   */
+  addressPrefix: asString,
+
+  /**
+   * ISO date string to start searching for transactions.
+   */
+  startIsoDate: asString,
+
+  /**
+   * ISO date string to end searching for transactions.
+   */
+  endIsoDate: asString
+})
+
+interface TxInfo {
+  providerId: string
+  orderId: string
+  isoDate: string
+  sourceAmount: number // exchangeAmount units
+  sourceCurrencyCode: string
+  sourcePluginId?: string
+  sourceTokenId?: EdgeTokenId
+  destinationAddress?: string
+  destinationAmount: number // exchangeAmount units
+  destinationPluginId?: string
+  destinationTokenId?: EdgeTokenId
+}
+
+export const getTxInfoRouter = Router()
+
+getTxInfoRouter.get('/', async function(req, res) {
+  const query = trial(
```

Comment:
trial seems like overkill. you basically use it to replace a single try/catch block.

Reply:
You can't return a value from a try-catch block.